storefront/pages/_app.tsx

```diff
@@ -9,16 +9,37 @@ import IntlWrapper from '../modules/i18n/components/IntlWrapper';
 import { CartContext } from '../modules/cart/CartContext';
 import withApollo from '../modules/apollo/utils/withApollo';
 import getMessages from '../modules/i18n/utils/getMessages';
+import ROUTES_CONFIG from '../modules/common/utils/getRouteConfig';
 
 const UnchainedApp = ({ Component, pageProps, router }) => {
   const messages = getMessages(router.locale);
+
   const toggleCart = (val) => {
     // eslint-disable-next-line no-use-before-define
     setCartContext({
       isCartOpen: val,
       toggleCart,
     });
   };
+  /* change root pages slug for there localy 
+  handles localization for routes /page | /page/1234
+  where 1234 is either id or slug */
+  useEffect(() => {
```

Comment:
Leave this up to nextjs, we shouldn't need to do this

Reply:
@pozylon nextjs doesn't detect slugs locale when changing for wildcard routes like [...slug]. if a person is looking at a product that is located say `/x/y/z/product-name` and decides to change to a different locale. browser URL is not changed by next and the slug is coming from the engine. 
that's why this is needed